The user defined in a Dockerfile can't create directories inside of a volume created during stack creation

**Describe the bug**
The user defined in a Dockerfile can't create directories inside of a volume created during stack creation. From a high-level analysis it seems like the volumes we create during deployment are owned by 'root'. While normally this is not an issue, this causes problems if a) the user of the main process is not root AND b) the user tries to create files or folders in said volume. A typical scenario for this is when creating a file for logging.

**To Reproduce**
Steps to reproduce the behavior:

1. Deploy https://github.com/codeforequity-at/botium-speech-processing
2. See the status of the frontend service. 

**Expected behavior**
The user defined in the dockerfile should be able to read/write inside the volumes we create.

**Desktop (please complete the following information):**

- OS: macOS
- Version 1.4.10

I tried what you said and still have these errors. You can try too maybe I'm doing it wrong. Updated Dockerfile https://github.com/Shubhamjugran/botium-speech-processing/tree/master/frontend 

2022-01-12 08:36:57.00 UTCfrontend-86bff98446-m4zfl[pod-event]Successfully pulled image "registry.cloud.okteto.net/voice-shubhamjugran/botium-speech-processing-frontend:okteto-with-volume-mounts" in 302.350138ms
2022-01-12 08:36:58.27 UTCfrontend-86bff98446-m4zflfrontend
2022-01-12 08:36:58.27 UTCfrontend-86bff98446-m4zflfrontend> botium-speech-processing-frontend@1.0.0 start-dist /app
2022-01-12 08:36:58.27 UTCfrontend-86bff98446-m4zflfrontend> node -r dotenv-flow/config ./src/server.js
2022-01-12 08:36:58.27 UTCfrontend-86bff98446-m4zflfrontend
2022-01-12 08:36:58.35 UTCfrontend-86bff98446-m4zflfrontenddotenv-flow: "BOTIUM_SPEECH_PROVIDER_TTS" is already defined in `process.env` and will not be overwritten
2022-01-12 08:36:58.35 UTCfrontend-86bff98446-m4zflfrontenddotenv-flow: "BOTIUM_SPEECH_PROVIDER_STT" is already defined in `process.env` and will not be overwritten
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend/app/node_modules/mkdirp/lib/mkdirp-native.js:35
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend throw er
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend ^
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontendError: EACCES: permission denied, mkdir '/app/resources/.cache/stt'
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Object.mkdirSync (fs.js:987:3)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at mkdirpNativeSync (/app/node_modules/mkdirp/lib/mkdirp-native.js:29:10)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Function.mkdirpSync [as sync] (/app/node_modules/mkdirp/index.js:21:7)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Object.<anonymous> (/app/src/routes.js:30:26)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Module._compile (internal/modules/cjs/loader.js:1063:30)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Module.load (internal/modules/cjs/loader.js:928:32)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Function.Module._load (internal/modules/cjs/loader.js:769:14)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at Module.require (internal/modules/cjs/loader.js:952:19)
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend at require (internal/modules/cjs/helpers.js:88:18) {
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend errno: -13,
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend syscall: 'mkdir',
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend code: 'EACCES',
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend path: '/app/resources/.cache/stt'
2022-01-12 08:36:59.31 UTCfrontend-86bff98446-m4zflfrontend}
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! code ELIFECYCLE
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! errno 1
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! botium-speech-processing-frontend@1.0.0 start-dist: `node -r dotenv-flow/config ./src/server.js`
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! Exit status 1
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! 
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! Failed at the botium-speech-processing-frontend@1.0.0 start-dist script.
2022-01-12 08:36:59.33 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2022-01-12 08:36:59.34 UTCfrontend-86bff98446-m4zflfrontend
2022-01-12 08:36:59.34 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! A complete log of this run can be found in:
2022-01-12 08:36:59.34 UTCfrontend-86bff98446-m4zflfrontendnpm ERR! /home/node/.npm/_logs/2022-01-12T08_36_59_338Z-debug.log

--- a/pkg/cmd/stack/translate.go
+++ b/pkg/cmd/stack/translate.go
@@ -277,7 +277,6 @@ func translateDeployment(svcName string, s *model.Stack) *appsv1.Deployment {
 	svc := s.Services[svcName]
 
 	healthcheckProbe := getSvcProbe(svc)
-
 	return &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        svcName,
@@ -300,6 +299,8 @@ func translateDeployment(svcName string, s *model.Stack) *appsv1.Deployment {
 				},
 				Spec: apiv1.PodSpec{
 					TerminationGracePeriodSeconds: pointer.Int64Ptr(svc.StopGracePeriod),
+					Volumes:                       translateVolumes(svcName, svc),
+					InitContainers:                getInitContainers(svcName, s),
 					Containers: []apiv1.Container{
 						{
 							Name:            svcName,
@@ -310,6 +311,7 @@ func translateDeployment(svcName string, s *model.Stack) *appsv1.Deployment {
 							Ports:           translateContainerPorts(svc),
 							SecurityContext: translateSecurityContext(svc),
 							Resources:       translateResources(svc),
+							VolumeMounts:    translateVolumeMounts(svcName, svc),
 							WorkingDir:      svc.Workdir,
 							ReadinessProbe:  healthcheckProbe,
 							LivenessProbe:   healthcheckProbe,
@@ -320,6 +322,7 @@ func translateDeployment(svcName string, s *model.Stack) *appsv1.Deployment {
 		},
 	}
 }
+
 func translatePersistentVolumeClaim(volumeName string, s *model.Stack) apiv1.PersistentVolumeClaim {
 	volumeSpec := s.Volumes[volumeName]
 	labels := translateVolumeLabels(volumeName, s)
@@ -450,15 +453,55 @@ func getInitContainers(svcName string, s *model.Stack) []apiv1.Container {
 	if len(svc.Volumes) > 0 {
 		addPermissionsContainer := getAddPermissionsInitContainer(svcName, svc)
 		initContainers = append(initContainers, addPermissionsContainer)
+		initializationContainer := getInitializeVolumeContentContainer(svcName, svc)
+		if initializationContainer != nil {
+			initContainers = append(initContainers, *initializationContainer)
+		}
 	}
-	initializationContainer := getInitializeVolumeContentContainer(svcName, svc)
-	if initializationContainer != nil {
-		initContainers = append(initContainers, *initializationContainer)
+
+	if len(svc.VolumeMounts) > 0 {
+		initializeVolumeMount := getInitializeVolumeMountsContainer(svcName, svc)
+		if initializeVolumeMount != nil {
+			initContainers = append(initContainers, *initializeVolumeMount)
+		}
 	}
 
 	return initContainers
 }
 
+func getInitializeVolumeMountsContainer(svcName string, svc *model.Service) *apiv1.Container {
+	initContainerCommand, initContainerVolumeMounts := getVolumeMountsCommandAndVolumeMounts(*svc)
+	if len(initContainerVolumeMounts) == 0 {
+		return nil
+	}
+	initContainer := &apiv1.Container{
+		Name:         fmt.Sprintf("init-%s", svcName),
+		Image:        "busybox",
+		Command:      initContainerCommand,
+		VolumeMounts: initContainerVolumeMounts,
+	}
+	return initContainer
+}
+
+func getVolumeMountsCommandAndVolumeMounts(svc model.Service) ([]string, []apiv1.VolumeMount) {
+	volumeMounts := []apiv1.VolumeMount{}
+
+	var command string
+	for idx, volume := range svc.VolumeMounts {
+		if fs, err := os.Stat(volume.LocalPath); err == nil && !fs.IsDir() {
+			continue
+		}
+		volumeName := fmt.Sprintf("volume-mount-%d", idx)
+		volumeMounts = append(volumeMounts, apiv1.VolumeMount{Name: volumeName, MountPath: volume.RemotePath})
+		newCommand := fmt.Sprintf("mkdir -p %s && chmod 777 %s", volume.RemotePath, volume.RemotePath)
+
+		if len(command) > 0 {
+			newCommand = fmt.Sprintf(" && %s", newCommand)
+		}
+		command += newCommand
+	}
+	return []string{"sh", "-c", command}, volumeMounts
+}
 func getAddPermissionsInitContainer(svcName string, svc *model.Service) apiv1.Container {
 	initContainerCommand, initContainerVolumeMounts := getInitContainerCommandAndVolumeMounts(*svc)
 	initContainer := apiv1.Container{
@@ -565,6 +608,21 @@ func translateVolumes(svcName string, svc *model.Service) []apiv1.Volume {
 			VolumeSource: apiv1.VolumeSource{
 				PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
 					ClaimName: volume.LocalPath,
+				},
+			},
+		})
+	}
+	for idx, v := range svc.VolumeMounts {
+		if fs, err := os.Stat(v.LocalPath); err == nil && !fs.IsDir() {
+			continue
+		}
+		name := fmt.Sprintf("volume-mount-%d", idx)
+		maxSize := resource.MustParse("1Gi")
+		volumes = append(volumes, apiv1.Volume{
+			Name: name,
+			VolumeSource: apiv1.VolumeSource{
+				EmptyDir: &apiv1.EmptyDirVolumeSource{
+					SizeLimit: &maxSize,
 				},
 			},
 		})
@@ -805,6 +863,19 @@ func translateVolumeMounts(svcName string, svc *model.Service) []apiv1.VolumeMou
 				MountPath: v.RemotePath,
 				Name:      name,
 				SubPath:   subpath,
+			},
+		)
+	}
+	for idx, v := range svc.VolumeMounts {
+		if fs, err := os.Stat(v.LocalPath); err == nil && !fs.IsDir() {
+			continue
+		}
+		name := fmt.Sprintf("volume-mount-%d", idx)
+		result = append(
+			result,
+			apiv1.VolumeMount{
+				MountPath: v.RemotePath,
+				Name:      name,
 			},
 		)
 	}

--- a/pkg/cmd/stack/translate_test.go
+++ b/pkg/cmd/stack/translate_test.go
@@ -1549,3 +1549,370 @@ func Test_translateAffinity(t *testing.T) {
 		})
 	}
 }
+
+func Test_translateInitContainers(t *testing.T) {
+	svcName := "test"
+	tmpFile, err := os.CreateTemp("", "test.conf")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	tests := []struct {
+		name          string
+		stack         *model.Stack
+		initContainer []apiv1.Container
+	}{
+		{
+			name: "deployment (only volume mounts)",
+			stack: &model.Stack{
+				Services: map[string]*model.Service{
+					svcName: {
+						VolumeMounts: []model.StackVolume{
+							{
+								LocalPath:  tmpFile.Name(),
+								RemotePath: "/this/is/a/file",
+							},
+							{
+								LocalPath:  tmpDir,
+								RemotePath: "/new/test/dir",
+							},
+						},
+					},
+				},
+			},
+			initContainer: []apiv1.Container{
+				{
+					Name:    fmt.Sprintf("init-%s", svcName),
+					Image:   "busybox",
+					Command: []string{"sh", "-c", "mkdir -p /new/test/dir && chmod 777 /new/test/dir"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume-mount-1",
+							MountPath: "/new/test/dir",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "statefulset (only volumes)",
+			stack: &model.Stack{
+				Services: map[string]*model.Service{
+					svcName: {
+						Volumes: []model.StackVolume{
+							{
+								LocalPath:  "volume",
+								RemotePath: "/this/is/a/",
+							},
+						},
+					},
+				},
+			},
+			initContainer: []apiv1.Container{
+				{
+					Name:    fmt.Sprintf("init-%s", svcName),
+					Image:   "busybox",
+					Command: []string{"sh", "-c", "chmod 777 /volumes/*"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume",
+							MountPath: "/volumes/volume",
+						},
+					},
+				},
+				{
+					Name:            fmt.Sprintf("init-volume-%s", svcName),
+					Image:           "",
+					ImagePullPolicy: apiv1.PullIfNotPresent,
+					Command:         []string{"sh", "-c", "echo initializing volume... && (cp -Rv /this/is/a//. /init-volume-0 || true)"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume",
+							MountPath: "/init-volume-0",
+							SubPath:   "volume",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "statefulset (volumes and volume mounts)",
+			stack: &model.Stack{
+				Services: map[string]*model.Service{
+					svcName: {
+						Volumes: []model.StackVolume{
+							{
+								LocalPath:  "volume",
+								RemotePath: "/this/is/a/",
+							},
+						},
+						VolumeMounts: []model.StackVolume{
+							{
+								LocalPath:  tmpFile.Name(),
+								RemotePath: "/this/is/a/file",
+							},
+							{
+								LocalPath:  tmpDir,
+								RemotePath: "/new/test/dir",
+							},
+						},
+					},
+				},
+			},
+			initContainer: []apiv1.Container{
+				{
+					Name:    fmt.Sprintf("init-%s", svcName),
+					Image:   "busybox",
+					Command: []string{"sh", "-c", "chmod 777 /volumes/*"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume",
+							MountPath: "/volumes/volume",
+						},
+					},
+				},
+				{
+					Name:            fmt.Sprintf("init-volume-%s", svcName),
+					Image:           "",
+					ImagePullPolicy: apiv1.PullIfNotPresent,
+					Command:         []string{"sh", "-c", "echo initializing volume... && (cp -Rv /this/is/a//. /init-volume-0 || true)"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume",
+							MountPath: "/init-volume-0",
+							SubPath:   "volume",
+						},
+					},
+				},
+				{
+					Name:    fmt.Sprintf("init-%s", svcName),
+					Image:   "busybox",
+					Command: []string{"sh", "-c", "mkdir -p /new/test/dir && chmod 777 /new/test/dir"},
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "volume-mount-1",
+							MountPath: "/new/test/dir",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := getInitContainers(svcName, tt.stack)
+			assert.Equal(t, tt.initContainer, result)
+		})
+	}
+}
+
+func Test_translateVolumes(t *testing.T) {
+	svcName := "test"
+	tmpFile, err := os.CreateTemp("", "test.conf")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	sizeLimit := resource.MustParse("1Gi")
+	tests := []struct {
+		name         string
+		svc          *model.Service
+		volumeMounts []apiv1.Volume
+	}{
+		{
+			name: "deployment (only volume mounts)",
+			svc: &model.Service{
+				VolumeMounts: []model.StackVolume{
+					{
+						LocalPath:  tmpFile.Name(),
+						RemotePath: "/this/is/a/file",
+					},
+					{
+						LocalPath:  tmpDir,
+						RemotePath: "/new/test/dir",
+					},
+				},
+			},
+			volumeMounts: []apiv1.Volume{
+				{
+					Name: "volume-mount-1",
+					VolumeSource: apiv1.VolumeSource{
+						EmptyDir: &apiv1.EmptyDirVolumeSource{
+							SizeLimit: &sizeLimit,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "statefulset (only volumes)",
+			svc: &model.Service{
+				Volumes: []model.StackVolume{
+					{
+						LocalPath:  "volume",
+						RemotePath: "/this/is/a/",
+					},
+				},
+			},
+			volumeMounts: []apiv1.Volume{
+				{
+					Name: "volume",
+					VolumeSource: apiv1.VolumeSource{
+						PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "volume",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "statefulset (volumes and volume mounts)",
+			svc: &model.Service{
+				Volumes: []model.StackVolume{
+					{
+						LocalPath:  "volume",
+						RemotePath: "/this/is/a/",
+					},
+				},
+				VolumeMounts: []model.StackVolume{
+					{
+						LocalPath:  tmpFile.Name(),
+						RemotePath: "/this/is/a/file",
+					},
+					{
+						LocalPath:  tmpDir,
+						RemotePath: "/new/test/dir",
+					},
+				},
+			},
+			volumeMounts: []apiv1.Volume{
+				{
+					Name: "volume",
+					VolumeSource: apiv1.VolumeSource{
+						PersistentVolumeClaim: &apiv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: "volume",
+						},
+					},
+				},
+				{
+					Name: "volume-mount-1",
+					VolumeSource: apiv1.VolumeSource{
+						EmptyDir: &apiv1.EmptyDirVolumeSource{
+							SizeLimit: &sizeLimit,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := translateVolumes(svcName, tt.svc)
+			assert.Equal(t, tt.volumeMounts, result)
+		})
+	}
+}
+
+func Test_translateVolumesMounts(t *testing.T) {
+	svcName := "test"
+	tmpFile, err := os.CreateTemp("", "test.conf")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatalf("failed to create dynamic manifest file: %s", err.Error())
+	}
+	tests := []struct {
+		name         string
+		svc          *model.Service
+		volumeMounts []apiv1.VolumeMount
+	}{
+		{
+			name: "deployment (only volume mounts)",
+			svc: &model.Service{
+				VolumeMounts: []model.StackVolume{
+					{
+						LocalPath:  tmpFile.Name(),
+						RemotePath: "/this/is/a/file",
+					},
+					{
+						LocalPath:  tmpDir,
+						RemotePath: "/new/test/dir",
+					},
+				},
+			},
+			volumeMounts: []apiv1.VolumeMount{
+				{
+					Name:      "volume-mount-1",
+					MountPath: "/new/test/dir",
+				},
+			},
+		},
+		{
+			name: "statefulset (only volumes)",
+			svc: &model.Service{
+				Volumes: []model.StackVolume{
+					{
+						LocalPath:  "volume",
+						RemotePath: "/this/is/a/",
+					},
+				},
+			},
+			volumeMounts: []apiv1.VolumeMount{
+				{
+					Name:      "volume",
+					MountPath: "/this/is/a/",
+					SubPath:   "volume",
+				},
+			},
+		},
+		{
+			name: "statefulset (volumes and volume mounts)",
+			svc: &model.Service{
+				Volumes: []model.StackVolume{
+					{
+						LocalPath:  "volume",
+						RemotePath: "/this/is/a/",
+					},
+				},
+				VolumeMounts: []model.StackVolume{
+					{
+						LocalPath:  tmpFile.Name(),
+						RemotePath: "/this/is/a/file",
+					},
+					{
+						LocalPath:  tmpDir,
+						RemotePath: "/new/test/dir",
+					},
+				},
+			},
+			volumeMounts: []apiv1.VolumeMount{
+				{
+					Name:      "volume",
+					MountPath: "/this/is/a/",
+					SubPath:   "volume",
+				},
+				{
+					Name:      "volume-mount-1",
+					MountPath: "/new/test/dir",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := translateVolumeMounts(svcName, tt.svc)
+			assert.Equal(t, tt.volumeMounts, result)
+		})
+	}
+}

--- a/pkg/cmd/stack/translate_test.go
+++ b/pkg/cmd/stack/translate_test.go
@@ -169,6 +169,12 @@ func Test_translateDeployment(t *testing.T) {
 					"annotation1": "value1",
 					"annotation2": "value2",
 				},
+				VolumeMounts: []model.StackVolume{
+					{
+						LocalPath:  "/usr",
+						RemotePath: "/app",
+					},
+				},
 				Image:           "image",
 				Replicas:        3,
 				StopGracePeriod: 20,
@@ -254,6 +260,14 @@ func Test_translateDeployment(t *testing.T) {
 	if !reflect.DeepEqual(c.Resources, apiv1.ResourceRequirements{}) {
 		t.Errorf("Wrong container.resources: '%v'", c.Resources)
 	}
+
+	volumeMounts := []apiv1.VolumeMount{
+		{
+			Name:      "volume-mount-0",
+			MountPath: "/app",
+		},
+	}
+	assert.Equal(t, volumeMounts, c.VolumeMounts)
 }
 
 func Test_translateStatefulSet(t *testing.T) {
@@ -288,7 +302,7 @@ func Test_translateStatefulSet(t *testing.T) {
 				CapAdd:  []apiv1.Capability{apiv1.Capability("CAP_ADD")},
 				CapDrop: []apiv1.Capability{apiv1.Capability("CAP_DROP")},
 
-				Volumes: []model.StackVolume{{RemotePath: "/volume1"}, {RemotePath: "/volume2"}},
+				Volumes: []model.StackVolume{{RemotePath: "/volume1"}, {LocalPath: "usr", RemotePath: "/volume2"}},
 				Resources: &model.StackResources{
 					Limits: model.ServiceResources{
 						CPU:    model.Quantity{Value: resource.MustParse("100m")},
@@ -309,14 +323,14 @@ func Test_translateStatefulSet(t *testing.T) {
 		t.Errorf("Wrong statefulset name: '%s'", result.Name)
 	}
 	labels := map[string]string{
-		"label1":                    "value1",
-		"label2":                    "value2",
-		model.StackNameLabel:        "stackName",
-		model.StackServiceNameLabel: "svcName",
+		"label1":                      "value1",
+		"label2":                      "value2",
+		model.StackNameLabel:          "stackName",
+		model.StackServiceNameLabel:   "svcName",
+		"stack.okteto.com/volume-usr": "true",
 	}
-	if !reflect.DeepEqual(result.Labels, labels) {
-		t.Errorf("Wrong statefulset labels: '%s'", result.Labels)
-	}
+	assert.Equal(t, labels, result.Labels)
+
 	annotations := map[string]string{
 		"annotation1": "value1",
 		"annotation2": "value2",
@@ -334,9 +348,7 @@ func Test_translateStatefulSet(t *testing.T) {
 	if !reflect.DeepEqual(result.Spec.Selector.MatchLabels, selector) {
 		t.Errorf("Wrong spec.selector: '%s'", result.Spec.Selector.MatchLabels)
 	}
-	if !reflect.DeepEqual(result.Spec.Template.Labels, labels) {
-		t.Errorf("Wrong spec.template.labels: '%s'", result.Spec.Template.Labels)
-	}
+	assert.Equal(t, labels, result.Spec.Template.Labels)
 	if !reflect.DeepEqual(result.Spec.Template.Annotations, annotations) {
 		t.Errorf("Wrong spec.template.annotations: '%s'", result.Spec.Template.Annotations)
 	}
@@ -346,17 +358,19 @@ func Test_translateStatefulSet(t *testing.T) {
 	initContainer := apiv1.Container{
 		Name:    fmt.Sprintf("init-%s", "svcName"),
 		Image:   "busybox",
-		Command: []string{"sh", "-c", "chmod 777 /data"},
+		Command: []string{"sh", "-c", "chmod 777 /data && chmod 777 /volumes/*"},
 		VolumeMounts: []apiv1.VolumeMount{
 			{
 				MountPath: "/data",
 				Name:      pvcName,
 			},
+			{
+				MountPath: "/volumes/usr",
+				Name:      "usr",
+			},
 		},
 	}
-	if !reflect.DeepEqual(result.Spec.Template.Spec.InitContainers[0], initContainer) {
-		t.Errorf("Wrong statefulset init container: '%v' but expected '%v'", result.Spec.Template.Spec.InitContainers[0], initContainer)
-	}
+	assert.Equal(t, initContainer, result.Spec.Template.Spec.InitContainers[0])
 	initVolumeContainer := apiv1.Container{
 		Name:            fmt.Sprintf("init-volume-%s", "svcName"),
 		Image:           "image",
@@ -370,14 +384,13 @@ func Test_translateStatefulSet(t *testing.T) {
 			},
 			{
 				MountPath: "/init-volume-1",
-				Name:      pvcName,
-				SubPath:   "data-1",
+				Name:      "usr",
+				SubPath:   "usr",
 			},
 		},
 	}
-	if !reflect.DeepEqual(result.Spec.Template.Spec.InitContainers[1], initVolumeContainer) {
-		t.Errorf("Wrong statefulset init container: '%v' but expected '%v'", result.Spec.Template.Spec.InitContainers[1], initVolumeContainer)
-	}
+	assert.Equal(t, initVolumeContainer, result.Spec.Template.Spec.InitContainers[1])
+
 	c := result.Spec.Template.Spec.Containers[0]
 	if c.Name != "svcName" {
 		t.Errorf("Wrong statefulset container.name: '%s'", c.Name)
@@ -425,13 +438,11 @@ func Test_translateStatefulSet(t *testing.T) {
 		},
 		{
 			MountPath: "/volume2",
-			Name:      pvcName,
-			SubPath:   "data-1",
+			Name:      "usr",
+			SubPath:   "usr",
 		},
 	}
-	if !reflect.DeepEqual(c.VolumeMounts, volumeMounts) {
-		t.Errorf("Wrong container.volume_mounts: '%v'", c.VolumeMounts)
-	}
+	assert.Equal(t, volumeMounts, c.VolumeMounts)
 
 	vct := result.Spec.VolumeClaimTemplates[0]
 	if vct.Name != pvcName {
@@ -455,6 +466,7 @@ func Test_translateStatefulSet(t *testing.T) {
 	if !reflect.DeepEqual(vct.Spec, volumeClaimTemplateSpec) {
 		t.Errorf("Wrong statefulset volume claim template: '%v'", vct.Spec)
 	}
+
 }
 
 func Test_translateJobWithoutVolumes(t *testing.T) {

